scipy/integrate/_quadrature.py

```diff
@@ -1035,3 +1039,228 @@ def newton_cotes(rn, equal=0):
+        message = ("Exception encountered when attempting vectorized call to "
+                   f"`func`: {e}. `func` should accept two-dimensional array "
+                   "with shape `(n_points, len(a))` and return an array with "
+                   "the integrand value at each of the `n_points` for better "
+                   "performance.")
+        warnings.warn(message, stacklevel=3)
+
+        def vfunc(x):
+            return np.apply_along_axis(func, axis=-1, arr=x)
+
+    n_points_int = np.int64(n_points)
+    if n_points != n_points_int:
+        message = "`n_points` must be an integer."
+        raise TypeError(message)
+
+    n_estimates_int = np.int64(n_estimates)
+    if n_estimates != n_estimates_int:
+        message = "`n_estimates` must be an integer."
+        raise TypeError(message)
+
+    if qrng is None:
+        qrng = stats.qmc.Halton(dim)
+    elif not isinstance(qrng, stats.qmc.QMCEngine):
+        message = "`qrng` must be an instance of scipy.stats.qmc.QMCEngine."
+        raise TypeError(message)
+
+    if qrng.d != a.shape[0]:
+        message = ("`qrng` must be initialized with dimensionality equal to "
+                   "the number of variables in `a`, i.e., "
+                   "`qrng.random().shape[-1]` must equal `a.shape[0]`.")
+        raise ValueError(message)
+
+    rng_seed = getattr(qrng, 'rng_seed', None)
+    rng = stats._qmc.check_random_state(rng_seed)
+
+    if log not in {True, False}:
+        message = "`log` must be boolean (`True` or `False`)."
+        raise TypeError(message)
+
+    return (vfunc, a, b, n_points_int, n_estimates_int, qrng, rng, log, stats)
+
+
+QMCQuadResult = namedtuple('QMCQuadResult', ['integral', 'standard_error'])
+
+
+def qmc_quad(func, a, b, *, n_points=1024, n_estimates=8, qrng=None,
+             log=False, args=None):
+    """
+    Compute an integral in N-dimensions using Quasi-Monte Carlo quadrature.
+
+    Parameters
+    ----------
+    func : callable
+        The integrand. Must accept a single arguments `x`, an array which
+        specifies the point at which to evaluate the integrand. For efficiency,
+        the function should be vectorized to compute the integrand for each
+        element an array of shape ``(n_points, n)``, where ``n`` is number of
+        variables.
+    a, b : array-like
+        One-dimensional arrays specifying the lower and upper integration
+        limits, respectively, of each of the ``n`` variables.
+    n_points, n_estimates: int, optional
+        One QMC sample of `n_points` (default: 256) points will be generated
+        by `qrng`, and `n_estimates` (default: 8) statistically independent
+        estimates of the integral will be produced. The total number of points
+        at which the integrand `func` will be evaluated is
+        ``n_points * n_estimates``. See Notes for details.
+    qrng : scipy.stats.qmc.QMCEngine, optional
+        An instance of the QMCEngine from which to sample QMC points.
```

Comment:
I think we should also link `QMCEngine` in the description. As we do for Halton or Generator. Or we just rely on the link to be on the argument type?

Reply:
Hmm even the argument type is not linking.
Thoughts on how to fix that?
![image](https://user-images.githubusercontent.com/6570539/188931668-d2d84dca-f4aa-4a80-9f77-b178ae5d90fb.png)